[W6.1h][W11-B2]Zhan Yu

ZY-Ang opened the pull request:
Completed LO-Interfaces. Pretty straightforward stuff.
Anyway with regards to @gabriellhs 's definition of a interface during the tutorial, "a **contract** between a provider and the consumer of the interface". I thought it was quite a bad word to use as contract implies quid pro quo. An interface feels more like slavery. 😅

ZY-Ang committed:
Add required files

For LO-Printable

ZY-Ang committed:
Update Printable Interface to specifications

For LO-Interfaces

ZY-Ang committed:
Add implementation for the data types

For LO-Interfaces

okkhoy commented on src/seedu/addressbook/data/person/Printable.java:
```diff
@@ -0,0 +1,8 @@
+    /**
+     * Returns a printable string representation of the data type of a person
+     */
+    String getPrintableString();
```
did you miss the second part of the LO? you need to add a `getPrintableString` in `Person` class to use the other getPrintableString methods

ZY-Ang replied:
Fixed 4462f6afca3e6188ff3eb818e05fbbcd6772a3c9

okkhoy commented:
@ZY-Ang 
Some comments added. Please ack & close the PR after reading comments.

ZY-Ang committed:
Update implementation to use getPrintableString in ReadOnlyPerson

For getting printable string format using the getAsTextShowAll() function within ReadOnlyPerson
